internal/ethapi/api.go

```diff
@@ -1010,6 +1010,59 @@ func (s *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.
 	return result, nil
 }
 
+func (s *BlockChainAPI) GetBlobSidecars(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) ([]map[string]interface{}, error) {
+	block, err := s.b.BlockByNumberOrHash(ctx, blockNrOrHash)
+	if block == nil || err != nil {
+		// When the block doesn't exist, the RPC method should return JSON null
+		// as per specification.
+		return nil, nil
+	}
+	blobSidecars, err := s.b.GetBlobSidecars(ctx, block.Hash())
+	if err != nil {
+		return nil, err
+	}
+	result := make([]map[string]interface{}, len(blobSidecars))
+	blobIndex := 0
+	for txIndex, tx := range block.Transactions() {
+		if tx.Type() == types.BlobTxType {
+			result[blobIndex] = marshalBlobSidecar(blobSidecars[blobIndex], block.Hash(), block.NumberU64(), tx, txIndex)
+			blobIndex++
+		}
+		if blobIndex >= len(blobSidecars) {
+			return nil, nil
```

Comment:
return nil, nil --> break

Reply:
`if blobIndex >= len(blobSidecars) {}`
it should never happen, otherwise there could be a bug, so better to print an error message and return nil, err